src/table/virtualized-table/Table.tsx

```diff
@@ -17,24 +16,23 @@ import getHeights from './utils/get-height';
 import useScrollbarWidth from './hooks/use-scrollbar-width';
 
 const Table = (props: TableProps) => {
-  const { rect, pageInfo, paginationNeeded } = props;
+  const { rect, pageInfo, tableData } = props;
+  const { totalsPosition, columns, paginationNeeded } = tableData;
   const { layout, theme, styling } = useContextSelector(TableContext, (value) => value.baseProps);
   const ref = useRef<HTMLDivElement>(null);
   const headerRef = useRef<VariableSizeList>(null);
   const totalsRef = useRef<VariableSizeList>(null);
   const bodyRef = useRef<BodyRef>(null);
   const innerForwardRef = useRef() as React.RefObject<HTMLDivElement>;
   const { xScrollbarWidth, yScrollbarWidth } = useScrollbarWidth(ref);
-  const totals = useMemo(() => getTotalPosition(layout), [layout]);
   const bodyStyle = useMemo<BodyStyle>(
     () => ({
       ...styling.body,
       background: theme.background.color ?? 'transparent',
     }),
     [layout, theme.name()] // eslint-disable-line react-hooks/exhaustive-deps
   );
-  const { headerRowHeight, bodyRowHeight, headerAndTotalsHeight } = getHeights(styling.head, bodyStyle, totals);
-  const columns = useMemo(() => getColumns(layout), [layout]);
+  const { headerRowHeight, bodyRowHeight, headerAndTotalsHeight } = getHeights(styling.head, bodyStyle, totalsPosition);
   const tableRect = useMemo(() => toTableRect(rect, paginationNeeded), [rect, paginationNeeded]);
```

Comment:
can't we move other stuff to `getTableData` function as well? like `tableRect`, `useColumnSize` and `useTableCount` ? seems like they somehow need some piece of layout data in their core and you already have layout in `getTableData`

Reply:
We "can" but they are only used by the virtual scroll table so I don't think there is much value in it.

Currently `TableData` is used by the virtual scroll table because some of the shared code between the two tables depends on it. If that would have not been the case I think I would have gone with a different solution.